volunteer/models/volunteer_shift.py

```diff
@@ -0,0 +1,202 @@
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import format_datetime
+from odoo.tools.translate import _
+
+from odoo.addons.base.models.res_partner import _tz_get
+
+
+class Shift(models.Model):
+    _name = "volunteer.shift"
+    _description = "Shift"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+
+    # General fields
+
+    name = fields.Char(required=True)
+    max_volunteer_nb = fields.Integer("Max Volunteer", required=True)
+    remaining_slots = fields.Integer(compute="_compute_remaining_slots")
+    is_one_day = fields.Boolean(compute="_compute_is_one_day")
+
+    # Stage fields
+
+    @api.model
+    def _default_stage_id(self):
+        Stage = self.env["volunteer.shift.stage"]
+        return Stage.search([("state", "=", "draft")], limit=1)
```

Comment:
I would suggest using the element in the data instead of doing a search, because this can brings trouble if user creates new stages:
```suggestion
        return self.env.ref("volunteer.stage_draft")
```

Reply:
Done